multibuildingdetector/loss/ssdtripletloss.py

```diff
@@ -48,9 +48,12 @@ def _filter_overlapping_bboxs(self, mb_boxs, mb_confs):
             indices = utils.non_maximum_suppression(box, self.nms_thresh)
 
             confs.append(conf[indices])
-            labels.append(label[indices])
+            if chainer.cuda.available:
+                labels.append(label[indices].get())
+            else:
+                labels.append(label[indices])
         confs = F.concat(confs, axis=0)
-        labels = self.xp.concatenate(labels)
+        labels = np.concatenate(labels)
```

Comment:
Can you explain why this always has to be a `numpy` array.

Reply:
Because we want to use it as a key inside a dict later on where we group the labels. Therefore we need to copy it from the gpu to memory. Afterwards it is an numpy array.

Reply:
Actually moving that logic to here, the peformance was doubled on my machine. 0.1 -> 0.2

Reply:
Maybe we should try using this https://docs.chainer.org/en/stable/reference/generated/chainer.functions.concat.html